rewrite-core/src/main/java/org/openrewrite/GitRemote.java

```diff
@@ -46,6 +47,34 @@ public enum Service {
         Unknown
     }
 
+    public static URI buildRemoteUrl(Service service, boolean ssh, String origin, String path) {
```

Comment:
Now all of the components are here at the moment, but another alternative could be an instance method for `toURI(boolean)` which could convert the cloneUrl if it were present OR take the remaining components and build up the correct URI.

Reply:
Yeah I was doubting if we need that, but might be useful down the line.

Reply:
I dont think we want to convert the clone url to be honest as it can come in in many forms (including scp style) and we want the output here to be a URI. Since the parser is already there, might as well just use the result directly

Reply:
For the SCP style URL, git internally just prefixes it with `ssh://`, so in theory we could do the same when we see the naked scp-style clone URL, but just always using the `buildRemoteUrl` works equally as well.

Reply:
I dont think `ssh://git@github.com:openrewrite/rewrite.git` is a valid url though
At least most parsers do not like it.